lib/mongo/retryable.rb

```diff
@@ -107,23 +100,29 @@ def read_with_one_retry(&block)
     # @return [ Result ] The result of the operation.
     #
     # @since 2.1.0
-    def write_with_retry(&block)
+    def write_with_retry
+      retried = false
```

Comment:
Great idea. Using a constant for the retries to be consistent with the read retries.